pypeit/spectrographs/p200_tspec.py

```diff
@@ -0,0 +1,269 @@
+            list:
+
+        """
+        return ['dispname']
+
+    def pypeit_file_keys(self):
+        """
+        Add additional columns to the file block of the PypeIt file
+
+        Returns:
+            list:
+
+        """
+        pypeit_keys = super(P200TSPECSpectrograph, self).pypeit_file_keys()
+        pypeit_keys += ['calib', 'comb_id', 'bkg_id']
+        return pypeit_keys
+
+    def check_frame_type(self, ftype, fitstbl, exprng=None):
+        """
+        Check for frames of the provided type.
+        """
+        good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)
+
+        if ftype in ['pinhole','bias']:
+            # No pinhole frames
+            return np.zeros(len(fitstbl), dtype=bool)
+        if ftype == 'dark':
+            return good_exp & (fitstbl['target'] == 'lamp_off')
+        if ftype == 'standard':
+            return good_exp & ((fitstbl['idname'] == 'object') | (fitstbl['idname'] == 'Object'))
+        if ftype in ['pixelflat', 'trace']:
+            return good_exp & (fitstbl['target'] == 'lamp_on')
+        if ftype in 'science':
+            return good_exp & ((fitstbl['idname'] == 'object') | (fitstbl['idname'] == 'Object'))
+        if ftype in ['arc', 'tilt']:
+            return good_exp & ((fitstbl['idname'] == 'object') | (fitstbl['idname'] == 'Object'))
+        return np.zeros(len(fitstbl), dtype=bool)
+
+
+    def bpm(self, filename, det, shape=None, msbias=None):
+        """
+        Override parent bpm function with BPM specific to X-Shooter VIS.
+
+        .. todo::
+            Allow for binning changes.
+
+        Parameters
+        ----------
+        det : int, REQUIRED
+        msbias : numpy.ndarray, required if the user wishes to generate a BPM based on a master bias
+        **null_kwargs:
+            Captured and never used
+
+        Returns
+        -------
+        bpix : ndarray
+          0 = ok; 1 = Mask
+
+        """
+        msgs.info("Custom bad pixel mask for TSPEC")
+        bpm_img = self.empty_bpm(filename, det, shape=shape)
+        # ToDo: Build a custom bad pixel mask.
+
+        return bpm_img
+
+    @property
+    def norders(self):
+        return 5
+
+    @property
+    def order_spat_pos(self):
+        ord_spat_pos = np.array([0.3096, 0.4863, 0.6406, 0.7813, 0.9424])
+        return ord_spat_pos
+
+    @property
+    def orders(self):
+        return np.arange(7, 2, -1, dtype=int)
+
+    @property
+    def spec_min_max(self):
+        spec_max = np.asarray([np.inf]*self.norders)
+        spec_min = np.asarray([1024, -np.inf, -np.inf, -np.inf, -np.inf])
+        return np.vstack((spec_min, spec_max))
+
+    def order_platescale(self, order_vec, binning=None):
+        """
+        NIRES has no binning
+
+        Args:
+            order_vec (np.ndarray):
+            binning (optional):
+
+        Returns:
+            np.ndarray:
+
+        """
+        norders = order_vec.size
+        return np.full(norders, 0.15)
+
+    @property
```

Comment:
We got rid of these @propertys. See e.g. gemini_gnirs spectrograph file.

Reply:
What do you mean we got rid of these properties. I can see this property in GNIRS and NIRES